rules/compose.smk

```diff
@@ -0,0 +1,262 @@
+    if co2_budget_values == "fraction":
+        inputs.update(
+            {
+                "eurostat": rules.retrieve_eurostat_balances.output["directory"],
+                "co2": rules.retrieve_ghg_emissions.output["csv"],
+            }
+        )
+
+    # Add brownfield inputs for non-first horizons
+    if foresight == "overnight" and len(horizons) > 1:
+        raise ValueError(
+            "Overnight optimization can only be run for a single planning horizon."
+        )
+
+    if horizon != horizons[0]:
+        # Not first horizon - need previous network
+        prev_horizon = horizons[horizons.index(horizon) - 1]
+
+        if foresight == "myopic":
+            # Myopic uses solved network from previous horizon
+            inputs["network_previous"] = RESULTS + f"networks/solved_{prev_horizon}.nc"
+        elif foresight == "perfect":
+            # Perfect foresight uses composed network from previous horizon
+            inputs["network_previous"] = resources(
+                f"networks/composed_{prev_horizon}.nc"
+            )
+        else:
+            raise ValueError(f"Invalid foresight type: {foresight}")
+
+    return inputs
+
+
+# Main composition rule - combines all network building steps
+rule compose_network:
+    input:
+        unpack(get_compose_inputs),
+    output:
+        resources("networks/composed_{horizon}.nc"),
+    params:
+        # Pass entire config sections using config_provider
+        foresight=config_provider("foresight"),
+        electricity=config_provider("electricity"),
+        sector=config_provider("sector"),
+        clustering=config_provider("clustering"),
+        clustering_temporal=config_provider("clustering", "temporal"),
+        existing_capacities=config_provider("existing_capacities"),
+        pypsa_eur=config_provider("pypsa_eur"),
+        renewable=config_provider("renewable"),
+        conventional=config_provider("conventional"),
+        costs=config_provider("costs"),
+        load=config_provider("load"),
+        lines=config_provider("lines"),
+        links=config_provider("links"),
+        industry=config_provider("industry"),
+        limited_heat_sources=config_provider(
+            "sector", "district_heating", "limited_heat_sources"
+        ),
+        # Direct parameters
+        countries=config_provider("countries"),
+        snapshots=config_provider("snapshots"),
+        drop_leap_day=config_provider("enable", "drop_leap_day"),
+        # Derived parameters
+        energy_totals_year=config_provider("energy", "energy_totals_year"),
+        # Parameters from add_existing_baseyear
+        baseyear=config_provider("planning_horizons"),
+        renewable_carriers=config_provider("electricity", "renewable_carriers"),
+        heat_pump_sources=config_provider("sector", "heat_pump_sources"),
+        # Brownfield settings (for myopic)
+        h2_retrofit=config_provider("sector", "H2_retrofit"),
```

Comment:
This setting also applies to overnight. Perhaps just remove the comments for the `params` section here?